dev/src/main/java/org/eclipse/codewind/intellij/ui/wizard/AddExistingProjectWizard.java

```diff
@@ -108,16 +112,39 @@ public void run() {
                         String language = subtypeInfo.id;
                         // call validate again with type and subtype hint
                         // allows it to run extension commands if defined for that type and subtype
+                        boolean validationOk = true;
                         if (subtypeInfo != null) {
-                            ProjectUtil.validateProject(name, path, typeInfo.getId() + ":" + subtypeInfo.id, connection.getConid(), indicator);
+                            try {
+                                ProjectUtil.validateProject(name, path, typeInfo.getId() + ":" + subtypeInfo.id, connection.getConid(), indicator);
+                            } catch (Exception e) {
+                                validationOk = false;
+                                CoreUtil.invokeLater(() -> {
+                                    int rc1 = CoreUtil.showYesNoDialog(message("ProjectValidationTitle"), message("ProjectValidationFailedContinueProjectBind", e.getMessage()));
+                                    if (rc1 == Messages.YES) {
+                                        continueWithBind(name, path, language, typeInfo, indicator);
+                                    } else {
+                                        doBind = false;
+                                    }
+                                });
+                            }
+                        }
+                        if (doBind && validationOk) {
+                            continueWithBind(name, path, language, typeInfo, indicator);
                         }
-                        ProjectUtil.bindProject(name, path, language, typeInfo.getId(), connection.getConid(), indicator);
                     }
                 } catch (Exception e) {
```

Comment:
Given this change, this try-catch block is covering much more code than it needs to.  It only needs to protect the call to `ProjectUtil.bindProject()` on line 109.

However, that call to `ProjectUtil.bindProject()` on line 109 needs to be changed to a call to `continueWithBind()`, otherwise the wizard never gets dismissed, and the  project doesn't get opened in IntelliJ, and so this try-catch becomes redundant and should be removed.

Reply:
Hi John, this new try-catch block accounts for project validation, not project bind.  I believe it is still necessary.  This is how the Eclipse plugin works too.  There is project validation on the 'second' page as well as the 'third' or last page.  In this latter case, it will prompt the user to select `yes` if they want to continue still bind the project regardless of validation issue, or `no` and they will remain on the page, where they can then choose another project type or language and then project validation will occur again (hence the need for project validation on this last page, even though, it happened on the second page with the original assessed project type).

Reply:
The call to bindProject on line 109 only happens when the project type is the same as the one that was detected earlier on the second page.  In this case, there is no need to validate the project again.

Reply:
Hi Keith, the try-catch I'm referring to is the old one, starting at line 101, and the only thing within it that can throw an exception is the `ProjectUtil.bindProject()` call on line 109.  And that call needs to be replaced with a call to `continueWithBind()` otherwise the wizard never gets dismissed and the bound project never gets opened.